src/economy.cpp

```diff
@@ -2004,6 +2004,7 @@ static void DoAcquireCompany(Company *c, bool hostile_takeover)
 	InvalidateWindowClassesData(WC_SHIPS_LIST, 0);
 	InvalidateWindowClassesData(WC_ROADVEH_LIST, 0);
 	InvalidateWindowClassesData(WC_AIRCRAFT_LIST, 0);
+	InvalidateWindowClassesData(WC_CLIENT_LIST);
```

Comment:
```suggestion
	InvalidateWindowData(WC_CLIENT_LIST, 0);
```
All other instances where `WC_CLIENT_LIST` is invalidated, it is done this way.

Regarding my other concern: I don't see any code path where the crashing bit of code is called without first checking the validity of the `company_id`. So it's only the server owner that can inflict this on themselves.